addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComLighting1Message.java

```diff
@@ -175,6 +199,55 @@ public String getDeviceId() {
     }
 
     @Override
+    public Command convertToCommand(RFXComValueSelector valueSelector) throws RFXComException {
+
+        Command OHcommand;
+
+        if (valueSelector.getItemClass() == SwitchItem.class) {
+
+            if (valueSelector == RFXComValueSelector.COMMAND) {
+
+                switch (command) {
+                    case OFF:
+                    case GROUP_OFF:
+                        OHcommand = OnOffType.OFF;
+                        break;
+
+                    case ON:
+                    case GROUP_ON:
+                        OHcommand = OnOffType.ON;
+                        break;
+
+                    case DIM:
+                        OHcommand = IncreaseDecreaseType.DECREASE;
```

Comment:
Yes, it works as expected. The valueSelector list is just used to derive a list of channel types the incoming message from the RFXCOM might apply to. It is passed to the convertTo*() methods so they can adapt their behaviour depending what channel they are being asked to feed. We're not actually feeding commands to or from a SwitchItem. It is entirely right for an INCREASE or DECREASE to be sent on the COMMAND channel. It is perfectly ok to have a switch item hooked up to that channel - it just has no interest in INCREASE/DECREASE. It is perfectly ok to have the same switch item hooked to a dimmer channel as well - the INCREASE/DECREASE command will propagate there and the dimmer will do its thing. It just works. It should also be possible to hook the switch item to a colour channel in which case the bright/dim keys should cycle through the colour wheel. But that only just occurred to me so I haven't tried it.